src/utils/are_codecs_compatible.ts

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright 2015 CANAL+ Group
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * This function is a shortcut that helps differentiate two codecs.
+ *
+ * @param codecA
+ * @param codecB
+ * @returns A boolean that tell whether or not those two codecs provided are even.
+ */
+function areCodecsCompatible(a: string, b: string): boolean {
+  const [mimeTypeA, ...propsA] = a.split(";");
+  const [mimeTypeB, ...propsB] = b.split(";");
+
+  if (mimeTypeA !== mimeTypeB) {
+    return false;
+  }
+
+  const codecsA = propsA.find((prop) => prop.startsWith("codecs="));
+  const codecsB = propsB.find((prop) => prop.startsWith("codecs="));
+  if (codecsA === undefined || codecsB === undefined) {
+    return false;
+  }
+
+  const [, codecA] = codecsA.split("=");
+  const [, codecB] = codecsB.split("=");
+
+  if (codecA.split(".")[0] !== codecB.split(".")[0]) {
```

Comment:
What if `codecA === ""`?
This code will throw as you're indexing without checking the length first.

Reply:
Nope, forget what I've written!
Reading more closely there's no big problem here